File: problems/0009_palindrome-number/reversed-int/solution.rb
#!/usr/bin/ruby

# @param {Integer} x
# @return {Boolean}
def is_palindrome(x)
    if x < 0
        return false
    elsif x < 10
        return true
    end
    reversed, k = [0, x]
    while k != 0
        reversed = reversed * 10 + (k % 10)
        k /= 10
    end
    return x == reversed
end

File: problems/0009_palindrome-number/reversed-int/solution.ts
function isPalindrome(x: number): boolean {
    if (x < 0)
        return false
    else if (x < 10)
        return true
    let reversed = 0
    for (let k = x; k != 0; k = Math.trunc(k / 10))
        reversed = reversed * 10 + (k % 10)
    return x == reversed
};
